### PR TITLE
[FEAT]: add the possibility to give an authorization

### PR DESCRIPTION
**Pull request by meriamBenSassi**
## Description

- Add the possibility to give a string Authorization.
- Add `partnerId` in the `PostBanksUserDto`.

## Motivation and Context

We would use this as the following in order to transfer directly the token used by the front. The `credentials.clientId` is a required property, In that case, it is only used for logs. Should we keep it?

Usage example: 
```ts
import { BanksUser, RequestBuilder } from '@algoan/rest';

/**
 * Init the  algoanRequest
**/
const algoanRequest: RequestBuilder = new RequestBuilder(config.algoan.baseUrl, {
    clientId: config.algoan.clientId,  // Not sure if we should keep this as required
});

/**
 * Create a new Banks User using the requestBuilder directy
 */
const response: BanksUser = await algoanRequest.request({
    method: 'POST',
    headers: { Authorization: "Bearer json web token" } // ferch this from the original request header
    url: '/v1/banks-users',
    data: {
        callbackUrl: `somecallback`,
        adenTriggers: {
            onSynchronizationFinished: true,
            bankreaderLinkRequired: true,
        },
    },
});

/**
 * Create a new Banks User using the exposed APIs
 */
algoanRequest.setAuthorizationHeader('Bearer json web token');
const banksUser = BanksUser.create(
      {
        partnerId: 'id123',
      },
      algoanRequest,
 );
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by qhdinh — APPROVED** [review]
The code looks good to me 👍  I just wonder the benefit of passing the token from the front directly without regenerating a new one. Regenerating a new one could minimize the risk that the current token is expired.

**Comment by meriamBenSassi** [comment]
> The code looks good to me 👍 I just wonder the benefit of passing the token from the front directly without regenerating a new one. Regenerating a new one could minimize the risk that the current token is expired.

There can be use cases where we need to use different credential according to the calling user. 

However, what is not covered by this PR is giving the token to the exposed API. (exemple `BanksUser.getBanksUserById` ) I'll try to do that.

**Review by qhdinh — APPROVED** [review]
Thanks @meriamBenSassi !

File: src/RequestBuilder.ts [hunk 1]
```diff
@@ -169,6 +173,14 @@ export class RequestBuilder {
...
     return response.data;
   }
+
+  /**
+   * Set the property authorizationHeader
+   * @param authorizationHeader : authorization to set
+   */
+  public setAuthorizationHeader(authorizationHeader: string): void {
+    this.authorizationHeader = authorizationHeader;
+  }
```
Comment (ccoeurderoy) [inline src/RequestBuilder.ts:183]: You could create a setter instead

```suggestion
  public set authorization(authorizationHeader: string) {
    this.authorizationHeader = authorizationHeader;
  }
```

**Review by ccoeurderoy — COMMENTED** [review]
LGTM! But I think that a setter could be cleaner than a classic public method 😉

File: test/request-builder.test.ts [hunk 2]
```diff
@@ -341,4 +341,101 @@ describe('Tests related to the RequestBuilder class', () => {
...
+
+    await requestBuilder.request({
+      method: 'GET',
+      url: '/',
+    });
+
+    expect(oAuthServer.isDone()).toBeFalsy();
+    expect(randomAlgoanServer.isDone()).toBeTruthy();
+    expect((requestBuilder as any).accessTokenInstance).toBeUndefined();
+    expect((requestBuilder as any).authorizationHeader).toBeDefined();
```
Comment (ccoeurderoy) [inline test/request-builder.test.ts:404]: I think you wanted to test the private property, not the public method right?

```suggestion
    expect((requestBuilder as any)._authorizationHeader).toBeDefined();
```

**Review by ccoeurderoy — CHANGES_REQUESTED** [review]

**Review by ccoeurderoy — APPROVED** [review]

**Pull request merged**